src/cartridges/int_adyen_overlay/cartridge/scripts/adyenGiving.js

```diff
@@ -42,14 +42,18 @@ function donate(donationReference, donationAmount, orderToken) {
     const donationToken = paymentInstrument.paymentTransaction.custom.Adyen_donationToken;
     const originalReference = paymentInstrument.paymentTransaction.custom.Adyen_pspReference;
     const paymentData = JSON.parse(paymentInstrument.paymentTransaction.custom.Adyen_log);
-    const paymentCurrency = paymentData.amount.currency;
+    const paymentCurrency = paymentData.amount.currency || paymentData.fullResponse?.amount?.currency;
     const availableDonationAmounts = AdyenHelper.getDonationAmounts();
-    paymentMethodVariant = paymentInstrument.custom.Adyen_Payment_Method_Variant;
+    paymentMethodVariant = paymentData.paymentMethod?.type || paymentData.fullResponse?.paymentMethod?.type;
 
     // for iDeal donations, the payment method variant needs to be set to sepadirectdebit
     if (paymentMethodVariant === 'ideal') {
       paymentMethodVariant = 'sepadirectdebit';
```

Comment:
as per the [api docs for donations](https://docs.adyen.com/api-explorer/Checkout/70/post/donations#request-paymentMethod-IdealDetails-type) the `paymentMethod.type` in requestObject should be 'ideal'. Why are we setting this to 'sepadirectdebit'?

Reply:
Seems that the api explorer is not correct for applepay and ideal.
More information can be found [here](https://docs.adyen.com/online-payments/donations/web-component/?tab=same_payment_method_1).